fix: support token blacklist feature in rest_auth

bartvanandel opened the pull request:
Add correct serializer when rest_framework_simplejwt.token_blacklist is detected, mirroring the approach taken by dj-rest-auth.

bartvanandel commented:
@tfranzel could you have a look, and enable the workflow that requires approval?

tfranzel commented:
Hi @bartvanandel,

sry for the delay. Could you please rebase your change because the the GH worker image got retired and rescheduling the actions won't make use of the new one.

bartvanandel commented:
@tfranzel Rebase done!

tfranzel commented:
Finally had some time to review. I think this looks good. However, I believe there is conditional missing. Shouldn't this also be dependant on `USE_JWT`?

https://github.com/iMerica/dj-rest-auth/blob/545fd1f4c9ae570216989a8f5a563dd938e038af/dj_rest_auth/views.py#L171

bartvanandel commented:
Yeah probably a good idea. One can have `SIMPLE_JWT` in their settings, but choose not to enable it for everything.

Updated and rebased on top of current master.

tfranzel commented:
@bartvanandel 

Thanks for getting back at me even though you moved on. Much appreciated. I wish you good luck with C# 😄 

> Posting an empty object vs None can mean 2 different things, no?

I guess so, but I don't think it matter much here. DRF might also accept `{}` for empty but I don't remember exactly either. In any case this patch is an improvement either way.

closed by 5221afd
